dashboard/src/actions/catalog.ts

```diff
@@ -211,7 +212,7 @@ export function getPlans() {
 }
 
 export function getCatalog(ns?: string) {
-  return async (dispatch: Dispatch<IStoreState>) => {
+  return async (dispatch: ThunkDispatch<IStoreState, null, Action>) => {
```

Comment:
I am not sure I understand `ThunkDispatch`: the other functions in this file also call several times to `dispatch`, what's the difference here?  Is it because of the nested calls?

Reply:
If we are dispatching a pure sync action creator, one that contains { type: foo}, you can use the `Dispatch` type (i.e dispatch(installed())), on the other hand, if you are dispatching an async reduxThunk action, i.e `getBindings` you need to use this type instead. This is because, in those cases the signature of the action contains the original dispatch and the getState method. This type signature, in theory allows you to inject those types to the action.

```
function incrementIfOdd() {
  return (dispatch, getState) => {
  ...
```

We could have avoided this by overloading Dispatch to allow not only `type` but also a function but it will defuse the purpose of being able to define what info and return types the async action has.

Reply:
Long story short, you can always use ThunkDispatch since it is an overloaded but using specifically `redux.Dispatch` when possible will allow you to find issues like this one https://github.com/kubeapps/kubeapps/pull/650/files#diff-67442faa85ea33f56c4ef70c57573277R17 in which we are not actually sending the action (note the parenthesis) and was caught thanks to the new typing.

Reply:
got it, thanks for the detailed explanation! I agree that's better to use the restrictive `Dispatch` instead that `ThunkDispatch` always.